### PR TITLE
[CELEBORN-1455] Remove improper configs from config template

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
To remove improper default configs.

### Why are the changes needed?
Many users won't change default configs in the template causing bad performance in test scenarios.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA.

### CODE REVIEW

**Review by RexXiong — APPROVED** [review]
LGTM!

File: conf/celeborn-defaults.conf.template [hunk 1]
```diff
@@ -16,14 +16,6 @@
...
-celeborn.rpc.io.clientThreads                    8
-celeborn.rpc.io.serverThreads                    8
-celeborn.rpc.io.numConnectionsPerPeer            2
-
-celeborn.push.io.numConnectionsPerPeer           2
-celeborn.worker.push.io.threads                  8
-
 celeborn.metrics.enabled                         true
 
 celeborn.worker.storage.dirs                     /mnt/disk1,/mnt/disk2,/mnt/disk3,/mnt/disk4,/mnt/disk5
```
Comment (SteNicholas) [inline conf/celeborn-defaults.conf.template:21]: Could this config add disktype in each disk to tell users follow the pattern?

**Comment by SteNicholas** [comment]
@FMX, could the configs `celeborn.master.ha.ratis.raft.server.snapshot.auto.trigger.enabled` and `celeborn.master.ha.ratis.raft.server.snapshot.auto.trigger.threshold` be removed from template, which value use the default value in template?

File: conf/celeborn-defaults.conf.template (hunk 1 above)
Comment (FMX) [reply conf/celeborn-defaults.conf.template:21]: NO. This is a config file. The demo of this configuration is in the README that shows how to set disktype for different dirs.

**Review by SteNicholas — APPROVED** [review]
LGTM. Thanks for updates.

**Pull request closed without merging**